sqlmesh/dbt/model.py

```diff
@@ -119,6 +119,14 @@ def _validate_partition_by(cls, v: t.Any) -> t.Union[t.List[str], t.Dict[str, t.
         },
     }
 
+    @property
+    def time_column(self) -> t.Optional[str]:
+        if self.time_column_:
+            return self.time_column_
+        if isinstance(self.partition_by, dict) and self.partition_by["data_type"] != "int64":
+            return self.partition_by["field"]
```

Comment:
I think we should also make sure that the incremental strategy is `insert_overwrite`.

Reply:
Yes, good idea.